climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+        self.intensity = intensity_new.tocsr()
+        self.fraction = self.intensity.copy()
+        self.fraction.data.fill(1.0)
+
+
+    @staticmethod
+    def _clean_firms_csv(csv_firms):
+        """Read and remove low confidence data from firms:
+            - MODIS: remove data where confidence values are lower than CLEAN_THRESH
+            - VIIRS: remove data where confidence values are set to low (keep
+                nominal and high values)
+
+        Parameters:
+            csv_firms: csv file of the FIRMS data or pd.DataFrame of FIRMS data
+
+        Returns:
+            pd.DataFrame
+        """
+        if isinstance(csv_firms, pd.DataFrame):
+            firms = csv_firms
+        else:
+            firms = pd.read_csv(csv_firms)
+        # Check for the type of instrument (MODIS vs VIIRS)
+        # Remove data with low confidence interval
+        # Uniformize the name of the birghtness columns between VIIRS and MODIS
+        temp = pd.DataFrame()
+        if 'instrument' in firms.columns:
+            if firms.instrument.any() == 'MODIS' or firms.instrument.any() == 'VIIRS':
+                firms_modis = firms.drop(firms[firms.instrument == 'VIIRS'].index)
+                firms_modis.confidence = np.array(
+                    list(map(int, firms_modis.confidence.values.tolist())))
+                firms_modis = firms_modis.drop(firms_modis[ \
+                    firms_modis.confidence < CLEAN_THRESH].index)
+                temp = firms_modis
+                firms_viirs = firms.drop(firms[firms.instrument == 'MODIS'].index)
+                if firms_viirs.size:
+                    firms_viirs = firms_viirs.drop(firms_viirs[firms_viirs.confidence == 'l'].index)
+                    firms_viirs = firms_viirs.rename(columns={'bright_ti4':'brightness'})
+                    temp = temp.append(firms_viirs, sort=True)
+                    temp = temp.drop(columns=['bright_ti4'])
+
+            firms = temp
+            firms = firms.reset_index()
+            firms = firms.drop(columns=['index'])
+
+        firms['iter_ev'] = np.ones(len(firms), bool)
+        firms['cons_id'] = np.zeros(len(firms), int) - 1
+        firms['event_id'] = np.zeros(len(firms), int)
+        firms['clus_id'] = np.zeros(len(firms), int) - 1
+        firms['datenum'] = np.array(str_to_date(firms['acq_date'].values))
+        return firms
+
+    @staticmethod
+    def _firms_resolution(firms):
+        """ Returns resolution of satellite used in FIRMS in degrees
+
+        Parameters:
+            firms (pd.DataFrame): FIRMS data
+
+        Returns:
+            float
+        """
+        # Resolution in km of the centroids depends on the data origin.
+        if 'instrument' in firms.columns:
+            if firms['instrument'].any() == 'MODIS':
+                res_data = 1.0
+            else:
+                res_data = 0.375 # For VIIRS data
+        else:
+            res_data = RES_DATA # For undefined data origin, defined by user
```

Comment:
RES_DATA is not defined by the user, though. It's hard-coded as a global variable in the beginning of the script. Does it make sense to have this default res as 1km? Or should it rather throw an error, saying that the source is not defined?

Reply:
I think it makes sense as a default, as MODIS data comes on 1 km resolution. But happy to discuss where to put the default variables

Reply:
But the MODIS data is covered in line 665.

Reply:
agree. I'm removing RES_DATA as a default. See ad1847df